Caltech-IPAC/firefly
# Firefly-94 Workspace with WebDAV servers not supporting depth 'Infinity'

https://jira.ipac.caltech.edu/browse/FIREFLY-94
https://jira.lsstcorp.org/browse/DM-13112 (firefly part)

Test builds:
IrsaViewer with the unchanged behavior:
https://irsawebdev9.ipac.caltech.edu/firefly-94_ws_irsa/irsaviewer/ 

IrsaViewer as it would be if WebDAV server did not support depth-'infinity'
https://irsawebdev9.ipac.caltech.edu/firefly-94_ws_noInfinity/irsaviewer/

The behavior is controlled by application property workspace.propfind.infinity (server and client, must be '__$workspace.propfind.infinity' in config). If this property is not specified, we assume that WebDAV server supports depth='infinity'. In this case the whole file tree is retrieved at once and passed to the client.

If workspace.propfind.infinity=false, only one level of workspace file tree is retrieved at a time. The tree is updated as user clicks on folders.

To test:
- Log in. Use 'Load Catalog' from workspace to browse workspace files. 
- Check that tables and images can be saved into workspace and read from workspace.
- Create subfolders with "Add subfolder" when saving. (You need to use 'Enter' for new path to be added to the workspace.)
